--- old/PKGBUILD
+++ new/PKGBUILD
@@ -9,12 +9,10 @@
 arch=('i686' 'x86_64')
 license=('GPL')
 url='http://mpv.io'
-depends=(
-  'ffmpeg' 'lcms2' 'libdvdread' 'libcdio-paranoia' 'libquvi' 'libgl' 'enca'
-  'libxinerama' 'mpg123' 'libxv' 'libxkbcommon' 'libva' 'lirc-utils' 'wayland'
-  'desktop-file-utils' 'hicolor-icon-theme' 'xdg-utils' 'lua' 'libdvdnav'
-  'libguess' 'portaudio'
-)
+depends=('ffmpeg' 'lcms2' 'libdvdread' 'libcdio-paranoia' 'libquvi' 'libgl'
+         'enca' 'libxinerama' 'mpg123' 'libxv' 'libxkbcommon' 'libva'
+         'desktop-file-utils' 'hicolor-icon-theme' 'xdg-utils' 'libdvdnav'
+         'libguess')
 makedepends=('mesa' 'python-docutils' 'ladspa')
 options=('!emptydirs' '!buildflags')
 install=mpv.install
@@ -36,7 +34,21 @@
     --enable-zsh-comp \
     --enable-libmpv-shared \
     --enable-cdda \
-    --enable-portaudio
+    --disable-portaudio \
+    --disable-libsmbclient \
+    --enable-xss \
+    --disable-libbluray \
+    --enable-alsa \
+    --disable-wayland \
+    --enable-vaapi-hwaccel \
+    --enable-vdpau-hwaccel \
+    --disable-tv \
+    --disable-tv-v4l2 \
+    --disable-libv4l2 \
+    --disable-pvr \
+    --disable-audio-input \
+    --disable-lua \
+
 
   ./waf build
 }
