--- old/PKGBUILD
+++ new/PKGBUILD
@@ -9,7 +9,7 @@
 arch=('i686' 'x86_64')
 url="http://www.postgresql.org/"
 license=('custom:PostgreSQL')
-makedepends=('krb5' 'libxml2' 'python2' 'perl' 'tcl>=8.6.0' 'openssl>=1.0.0')
+makedepends=('python2' 'openssl>=1.0.0')
 source=(http://ftp.postgresql.org/pub/source/v${pkgver}/postgresql-${pkgver}.tar.bz2
         postgresql-run-socket.patch
         postgresql.pam postgresql.logrotate
@@ -29,33 +29,38 @@
             '7e086d70e0dcaa6ce45693b4e826ce92d770192b3aff5642319b1ef279d88dc4'
             'b3ed31e5882e5bdf4515d488d3f2db874ff5561be3091eab29810f831038a888')
 
-build() {
+prepare() {
   cd "${srcdir}/postgresql-${pkgver}"
 
   patch -Np1 < ../postgresql-run-socket.patch
+}
+
+build() {
+  cd "${srcdir}/postgresql-${pkgver}"
 
   ./configure --prefix=/usr \
   --mandir=/usr/share/man \
   --datadir=/usr/share/postgresql \
   --sysconfdir=/etc \
-  --with-krb5 \
-  --with-gssapi \
-  --with-libxml \
+  --without-krb5 \
+  --without-gssapi \
+  --without-libxml \
   --with-openssl \
-  --with-perl \
+  --without-perl \
   --with-python PYTHON=/usr/bin/python2 \
-  --with-tcl \
+  --without-tcl \
   --with-pam \
   --with-system-tzdata=/usr/share/zoneinfo \
-  --enable-nls \
-  --enable-thread-safety
+  --disable-nls \
+  --without-ldap \
+  --with-zlib \
 
   make world
 }
 
 package_postgresql-libs() {
   pkgdesc="Libraries for use with PostgreSQL"
-  depends=('krb5' 'openssl>=1.0.0' 'readline>=6.0')
+  depends=('openssl>=1.0.0' 'readline>=6.0')
   provides=('postgresql-client')
   conflicts=('postgresql-client')
 
@@ -114,7 +119,7 @@
 package_postgresql() {
   pkgdesc="A sophisticated object-relational DBMS"
   backup=('etc/pam.d/postgresql' 'etc/logrotate.d/postgresql')
-  depends=("postgresql-libs>=${pkgver}" 'krb5' 'libxml2' 'readline>=6.0' 'openssl>=1.0.0')
+  depends=("postgresql-libs>=${pkgver}" 'readline>=6.0' 'openssl>=1.0.0')
   optdepends=('python2: for PL/Python support'
               'perl: for PL/Perl support'
               'tcl: for PL/Tcl support'
